### PR TITLE
Fixed ignore _method

### PR DESCRIPTION
**Pull request by assurrussa**
## Description

Fixed a problem with the delete method when submitting via the form.

### CODE REVIEW

**Comment by tarampampam** [comment]
@assurrussa can you describe the problem with more details? how it can be reproduced?

**Comment by assurrussa** [comment]
> @assurrussa can you describe the problem with more details? how it can be reproduced?

When sending any form with a `DELETE` method or a `PATCH`, for example, the application ignores the hidden field `_method`
and as a result the action necessary for the form does not occur.

```
// route
Route::delete('/delete', 'AnyController@delete');
// form
<form action="/delete" method="POST">
    <input type="hidden" name="_method" value="DELETE">
    // ...
    <button type="submit">Submit</button> // Click Submit
</form>
```
In the `vendor/symfony/http-foundation/Request.php` file on line 1251, this request is being processed.
image

**This hotfix resolves the current issue.**

**Comment by tarampampam** [comment]
Ok, I understand you. In this case - maybe you will create a separate listener for this feature? Like `Spiral\RoadRunnerLaravel\Listeners\EnableHttpMethodParameterOverride` with something like that:

```php
class EnableHttpMethodParameterOverride implements ListenerInterface
{
    /**
     * {@inheritdoc}
     */
    public function handle($event): void
    {
        if ($event instanceof WithHttpRequest) {
            $event->httpRequest()->enableHttpMethodParameterOverride();
        }
    }
}
```

And include this listener into package configuration (`roadrunner.php`) before or after `RebindRouterListener`?

**Comment by tarampampam** [comment]
Also you should merge changes from `master` branch (7244e42b4398ef6c7b10d860cb4893fc42a6b991) into yours

**Comment by assurrussa** [comment]
@tarampampam I refactored the code for your project, please check

**Comment by tarampampam** [comment]
Looks great! Now you can write unit-tests for your code and add new record into `CHANGELOG.md` file

File: src/Listeners/EnableHttpMethodParameterOverride.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spiral\RoadRunnerLaravel\Listeners;
+
+use Spiral\RoadRunnerLaravel\Events\Contracts\WithHttpRequest;
+
+/**
+ * Class EnableHttpMethodParameterOverride
```
Comment (tarampampam) [inline src/Listeners/EnableHttpMethodParameterOverride.php:10]:
```suggestion
```

File: src/Listeners/EnableHttpMethodParameterOverride.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * This listener must be registered BEFORE `RebindRouterListener` if you need to process
+ * the hidden `_hidden` field when submitted using the HTML form.
+ *
+ * @see RebindRouterListener
+ * @package Spiral\RoadRunnerLaravel\Listeners
```
Comment (tarampampam) [inline src/Listeners/EnableHttpMethodParameterOverride.php:15]:
```suggestion
```

**Comment by tarampampam** [comment]
:+1: Thx for your contribution!

**Comment by tarampampam** [comment]
Released: https://github.com/spiral/roadrunner-laravel/releases/tag/v3.5.0

**Comment by tarampampam** [comment]
I forgot to rename class `EnableHttpMethodParameterOverride` to `EnableHttpMethodParameterOverrideListener`, my mistake. Release was updated with this commit: https://github.com/spiral/roadrunner-laravel/commit/4f3d2530b7e51ced9a4fd7473c5a8312054ea4e7

**Comment by assurrussa** [comment]
@tarampampam Thanks for the release, I also did not notice the spelling of the class is incorrect, thanks for editing.

**Pull request merged**